flow/project.py

```diff
@@ -680,6 +684,7 @@ def __call__(self, func):
             func._flow_groups.append(self.name)
         else:
             func._flow_groups = [self.name]
+        self.operations.append(func.__name__)
```

Comment:
There are two distinct problems here, both of which have roughly the same solution:
1. Anonymous functions (lambdas) have a `__name__`, it is just implicitly set to "\<lambda\>". The issue of naming conflicts is not unique to anonymous functions, it occurs any time that two operations have the same `__name__`. For example, if a users import operation functions from separate modules they could conflict e.g. `import foo; import bar; ... Project.operation(foo.baz), Project.operation(bar.baz)`. The handle we provide to users to circumvent this issue is naming the operation via that `name` parameter `Project.operation(foo.baz, "foo_baz"); Project.operation(bar.baz, "bar_baz")`, but it falls to users to make that distinction if necessary. Users wishing to use multiple anonymous functions have the same requirement.
2. Objects that actually don't have a `__name__` will fail [here](https://github.com/glotzerlab/signac-flow/blob/master/flow/project.py#L1415). This issue is possible because instances of `functools.partial`, for example, do not have a name: 
```
>>> x = partial(sum, [1, 2, 3])
>>> x.__name__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'functools.partial' object has no attribute '__name__'
```
I know @b-butler jumped through quite a few hoops in parts of the groups code to enable partials in certain cases. I would be fine requiring users to provide a name in that case as well, probably by wrapping the line I linked above with something like
```
try:
    name = func.__name__
except AttributeError:
    raise ValueError("A name could not be inferred from the passed callable, please provide the ``name`` parameter to ``Project.operation``.")

Reply:
@kidrahahjo I think enforcing this requirement explicitly is the easiest way for you to move forward.

Reply:
I added tests that raise `FlowProjectDefinitionError` for anonymous functions added to groups without first being added as operations. Resolving.